app/src/main/java/com/github/se/travelpouch/model/profile/ProfileModelView.kt

```diff
@@ -18,19 +17,21 @@ import kotlinx.coroutines.flow.asStateFlow
  * @param repository (ProfileRepository) : the repository that is used as a logic between Firebase
  *   and profiles
  */
-class ProfileModelView(private val repository: ProfileRepository) : ViewModel() {
+@HiltViewModel
+class ProfileModelView @Inject constructor(private val repository: ProfileRepository) :
+    ViewModel() {
 
   private val onFailureTag = "ProfileViewModel"
 
-  companion object {
-    val Factory: ViewModelProvider.Factory =
-        object : ViewModelProvider.Factory {
-          @Suppress("UNCHECKED_CAST")
-          override fun <T : ViewModel> create(modelClass: Class<T>): T {
```

Comment:
Now the factory is useless since we are using hilt injection. So at the creation of the view model, we use injection, which can take two different values depending of if we use tests or app